### PR TITLE
feat: replit support

### PR DESCRIPTION
**Pull request by Kodylow**
With Replit's new Nix Persistence feature enabled, this PR make Fedimint's developer environment work in Replit using the nix flake and mprocs. 

Fork Fedimint into a Repl, it automatically start you in the nix flake devShell, hitting "Run" start the mprocs dev environment

Special thanks to @ryantm !

image

### CODE REVIEW

File: .envrc [hunk 1]
```diff
@@ -1 +1,4 @@
+if ! has nix_direnv_version || ! nix_direnv_version 3.0.4; then
```
Comment (dpc) [inline .envrc:1]: Please add a comment explaining why is this needed.

File: .replit [hunk 2]
```diff
@@ -1,47 +1,15 @@
...
-# # Problem: `nix develop` too large
-#
-# There's too many dependencies in default shell, so I created a minimal `.#replit` dev shell.
-#
-# With this `bidgen` works.
-run = "nix --extra-experimental-features nix-command --extra-experimental-features flakes develop .#replit -c cargo check"
+# Replit support 
+run = "direnv allow && nix develop --command just mprocs"
 hidden = ["target"]
+entrypoint = "src/main.rs"
```
Comment (dpc) [inline .replit:4]: We don't have a main `src/main.rs` file.

**Review by elsirion — APPROVED** [review]

**Comment by elsirion** [comment]
After this PR things will likely be less broken than before, but if you have the time documenting what options do/why they are set would be great. Also, is this replit integration useful enough that we should have a badge for it? https://docs.replit.com/programming-ide/using-git-on-replit/running-github-repositories-replit#adding-a-run-on-replit-badge

File: .envrc (hunk 1 above)
Comment (Kodylow) [reply .envrc:1]: Replit needs that check for direnv to get picked up when you reconnect, if it's not there when you reconnect in browser then it kicks you out of the nix develop shell and you have to run `nix develop` again, not a big deal if I remove

File: .replit (hunk 2 above)
Comment (Kodylow) [reply .replit:4]: Entrypoint is just the first file shown when you open the repl, changing it to our README

File: .envrc (hunk 1 above)
Comment (elsirion) [reply .envrc:1]: I think just adding this comment in the code would go a long way re maintainability.

**Comment by Kodylow** [comment]
We could add the run with replit badge after nix persistence gets generally launched, for now it requires a special flag to use nix flakes. The normal replit.nix works now but once we have the flake option then when someone forks fedimint on replit it will start with the built nix artifacts for the flake on master so getting into the develop devShell takes like 10 seconds which will be super useful.

File: .envrc (hunk 1 above)
Comment (dpc) [reply .envrc:1]: Yeah, I don't think it hurts to have it here, but we just need a comment why it is there, otherwise it looks odd, unnecessary and even suspicious.

File: .config/bashrc [hunk 3]
```diff
@@ -0,0 +1 @@
+eval "$(direnv hook bash)"
```
Comment (dpc) [inline .config/bashrc:1]: While at it, please add a comment here as well on top, that the whole thing is for replit support.

File: .replit [hunk 4]
```diff
@@ -53,11 +21,15 @@ packageSearch = true
...
 # this must be kept in sync with `nixpkgs` channel in `flake.nix`
-channel = "stable-22_11"
+channel = "stable-23_11"
 
 [gitHubImport]
 requiredFiles = [".replit", "replit.nix"]
+
+[[ports]]
+localPort = 8333
+externalPort = 80
```
[the remark this answers is not in the document]
Comment (Kodylow) [reply .replit]: they're for deployments but we can't deploy currently since there's no way to persist rocksdb, removing.

**Review by dpc — APPROVED** [review]

File: .gitignore [hunk 5]
```diff
@@ -8,3 +8,7 @@
 .direnv
 .vscode
 .DS_Store
+
+.cargo/registry
+.cargo/.package-cache
+.cargo/.package-cache-mutate
```
[the remark this answers is not in the document]
Comment (Kodylow) [reply .gitignore]: added it, our git hooks throw weird errors in replit so have to --no-verify, I logged the git issues with them

**Review by dpc — APPROVED** [review]

File: .gitignore (hunk 5 above)
[the remark this answers is not in the document]
Comment (dpc) [reply .gitignore]: Well, I'm more thinking about long term solution so replit contributors get the text editor to do the right thing automatically. Possibly replit supports `.editconfig` or something?

[the remark this answers is not in the document]
Comment (dpc) [reply .gitignore]: https://editorconfig.org/

File: replit.nix [hunk 6]
```diff
@@ -1,5 +1,11 @@
...
+    env = {
+        OPENSSL_DIR = "${pkgs.openssl.dev}";
+        LIBCLANG_PATH = "${pkgs.libclang.lib}/lib";
+      };
   deps = [
+    pkgs.direnv
+    pkgs.just
     pkgs.gcc
     pkgs.mold
     pkgs.rustc
```
Comment (elsirion) [inline replit.nix:11]: This looks rather redundant, why can't we use our normal shell in replit?

**Review by elsirion — APPROVED** [review]

File: replit.nix (hunk 6 above)
Comment (Kodylow) [reply replit.nix:11]: This lets you build it outside nix develop without throwing an error when the nix persistence flag is enabled. If you have nix persistence enabled it ignores the replit.nix and just uses the flake
